src/Extensions/Microsoft.Diagnostics.Monitoring.Extension.Common/EgressHelper.cs

```diff
@@ -83,13 +71,69 @@ internal sealed class EgressHelper
                 result.FailureMessage = ex.Message;
             }
 
+            return ProcessEgressResult(result);
+        }
+
+        private static async Task<int> Validate<TProvider, TOptions>(Action<IServiceCollection> configureServices, CancellationToken token)
+            where TProvider : EgressProvider<TOptions>
+            where TOptions : class, new()
+        {
+            EgressArtifactResult result = new();
+            try
+            {
+                ExtensionEgressPayload configPayload = await GetPayload(token);
+                _ = await GetProviderAndOptions<TProvider, TOptions>(configureServices, configPayload);
+
+                result.ArtifactPath = string.Empty;
+                result.Succeeded = true;
+            }
+            catch (Exception ex)
+            {
+                result.Succeeded = false;
+                result.FailureMessage = ex.Message;
+            }
+
+            return ProcessEgressResult(result);
+        }
+
+        private static int ProcessEgressResult(EgressArtifactResult result)
+        {
             string jsonBlob = JsonSerializer.Serialize(result);
             Console.Write(jsonBlob);
-
             // return non-zero exit code when failed
             return result.Succeeded ? 0 : 1;
         }
 
+        private static async Task<(TOptions, EgressProvider<TOptions>)> GetProviderAndOptions<TProvider, TOptions>(Action<IServiceCollection> configureServices, ExtensionEgressPayload configPayload)
```

Comment:
I'm not certain that this refactor makes sense, given that the validation path doesn't involve the egress provider implementation.